spring-kafka/src/main/java/com/baeldung/partitioningstrategy/KafkaProducer.java

```diff
@@ -0,0 +1,26 @@
+package com.baeldung.partitioningstrategy;
+
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Service;
+
+@Service
+public class KafkaProducer {
+
+    private final KafkaTemplate<String, String> kafkaTemplate;
+
+    public KafkaProducer(KafkaTemplate<String, String> kafkaTemplate) {
```

Comment:
For clarity, do you think we should annotate this constructor with `@AutoWired` even though it is not strictly necessary?

Reply:
I've removed the wrapper.